lib/browser/tag/update.js

```diff
@@ -160,6 +161,39 @@ function updateTagRef(expr, parent) {
   expr.tag.update()
 }
 
+/**
+ * Update dynamically created riot-tag with changing expressions
+ * @param   { Object } expr - expression tag and expression info
+ * @param   { Tag } parent - parent for tag creation
+ */
+
+function updateRtag(expr, parent) {
+  var tagName = tmpl(expr.value, parent),
+    conf
+
+  if (expr.tag && expr.tagName == tagName) {
+    expr.tag.update()
+    return
+  }
+
+  // sync _parent to accommodate changing tagnames
+  if (expr.tag) {
+    var delName = expr.tag.opts.riotTag,
+      tags = expr.tag._parent.tags[delName]
+
+    if (isArray(tags)) tags.splice(tags.indexOf(expr.tag), 1); else delete expr.tag._parent.tags[delName]
+
+  }
+
```

Comment:
I added the code above, the issue was when a tag changed name, the _parent kept a copy and this got re-inherited.  I think this may be just a symptom of the fact that we never had tags that could morph to another tag before, but if somebody is more familiar with the _parent property and sees any other potential issues it would be appreciated, if not this is probably ok